app/models/call_result.rb

```diff
@@ -0,0 +1,24 @@
+class CallResult < ApplicationRecord
+  include Mergeable
+  belongs_to :appointment, optional: true
```

Comment:
This one's iffy. I realize we can't make it required because the appointment might reach us later. Can we validate the presence of `appointment_id` though? I'd love for our models to follow our API's contract as closely as possible.

Reply:
Not sure I follow this -- wouldn't the call result and appointment JSON come in the same payload from the client?

Reply:
@harimohanraj89 Can't do it cleanly, if we validate presence of appointment_id, we'd have to stop spec-ing the optional association (`it { should belong_to(:appointment).optional }`). Can't do both since the specs are contradictory.

We actually enforce this at the db instead, the column is required. This is how we generally model all our synced resources. 

@rsanheim Appointments and CallResults are synced separately, we're setting up a new sync API end point for CallResults.

Reply:
Hmm. I don't want to let shoulda-matchers dictate our implementation too much. The two specs aren't really contradictory, we want to say - you can have an appt id that points to nothing, but you can't have a blank appt id, right? I'd rather handroll that spec and rely on application-level validations, rather than letting it get down to the DB and raise an exception.